pkg/vm/builtins/builtin_runner.go

```diff
@@ -29,3 +33,25 @@ func Runner(name starknetParser.Builtin) memory.BuiltinRunner {
 		panic("Unknown builtin")
 	}
 }
+
+// GetAllocatedInstances calculates the number of instances of given builtin for current step and builtin ratio.
+// Ratio parameter defines the ratio between the number of steps to the number of builtin instances. It means that this builtin is expected to be used once every eight steps of the execution
```

Comment:
once every x steps (instead of eight?) , this method is generic for all builtins

Reply:
yes, thats right!